src/Core/Logging/Extensions/ILogServiceExtensions.Trace.cs

```diff
@@ -0,0 +1,166 @@
+            [CallerMemberName] string methodName = "")
+        {
+            service.Internal(
+                timestamp,
+                state,
+                methodName);
+        }
+
+        public static void Trace(
+            this ILogService service,
+            DateTimeOffset timestamp,
+            Exception ex,
+            [CallerMemberName] string methodName = "")
+        {
+            service.Internal(
+                timestamp,
+                ex,
+                methodName);
+        }
+
+
+
+        public static void Trace<TState>(
+            this ILogService service,
+            TState message,
+            Exception ex,
+            [CallerMemberName] string methodName = "")
+        {
+            service.Internal(
+                message,
+                ex,
+                methodName);
+        }
+
+        public static void Trace<TState>(
+            this ILogService service,
+            TState message,
+            [CallerMemberName] string methodName = "")
+        {
+            service.Internal(
+                message,
+                methodName);
+        }
+
+        public static void Trace(
+            this ILogService service,
+            Exception ex,
+            [CallerMemberName] string methodName = "")
+        {
+            service.Internal(
+                ex,
+                methodName);
+        }
+
+
+
+        private static void Trace<TState>(
+            this ILogService service,
+            TState message,
+            Exception ex,
+            Func<TState, Exception, string> formatter,
+            [CallerMemberName] string methodName = "")
+        {
+            service.Internal(
+                message,
+                ex,
+                formatter,
+                methodName);
+        }
+
+        private static void Trace<TState>(
+            this ILogService service,
+            TState message,
+            Func<TState, Exception, string> formatter,
+            [CallerMemberName] string methodName = "")
+        {
+            service.Internal(
+                message,
+                formatter,
+                methodName);
+        }
+
+        private static void Trace(
+            this ILogService service,
+            Exception ex,
+            Func<object, Exception, string> formatter,
+            [CallerMemberName] string methodName = "")
+        {
+            service.Internal(
+                ex,
+                formatter,
+                methodName);
+        }
+
+
+
+        private static void Trace<TState>(
+            this ILogService service,
+            DateTimeOffset timestamp,
+            TState message,
+            Func<TState, Exception, string> formatter,
+            [CallerMemberName] string methodName = "")
+        {
+            service.Internal(
+                timestamp,
+                message,
+                formatter,
+                methodName);
+        }
+
+        private static void Trace(
```

Comment:
Codacy found an issue: [Method has 5 parameters, which is greater than the 4 authorized.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5762132)